src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java

```diff
@@ -0,0 +1,115 @@
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.beanstalk.maven.plugin;
+
+import java.io.File;
+import java.util.zip.ZipFile;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.yaml.snakeyaml.error.YAMLException;
+
+/**
+ * Test case for {@link AbstractBeanstalkMojo}.
+ * @author Neo Matrix (Neo.matrix@gmail.com)
+ * @version $Id$
+ */
+public final class GenericMojoTest {
+    /**
+     * Test {@link AbstractBeanstalkMojo#execute()} method.
+     * @throws Exception If something is wrong.
+     * @checkstyle ExecutableStatementCountCheck (40 lines)
+     */
+    @Test
+    public void testExecute() throws Exception {
+        // @checkstyle IllegalTypeCheck (2 lines)
+        final AbstractBeanstalkMojo mojo =
+            Mockito.mock(AbstractBeanstalkMojo.class);
+        final File mockFile = Mockito.mock(File.class);
+        final ZipFile mockZipFile = Mockito.mock(ZipFile.class);
+        Mockito.when(mockFile.exists()).thenReturn(true);
+        Mockito.when(mojo.createZipFile()).thenReturn(mockZipFile);
+        Mockito.doCallRealMethod().when(mojo).execute();
+        Mockito.doCallRealMethod().when(mojo).getLog();
+        Mockito.doCallRealMethod().when(mojo)
+            .setWar(Mockito.any(File.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setName(Mockito.any(String.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setBucket(Mockito.any(String.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setKey(Mockito.any(String.class));
+        mojo.setWar(mockFile);
+        mojo.setName("name");
+        mojo.setBucket("bucket");
+        mojo.setKey("key");
+        mojo.execute();
+        Mockito.verify(mojo).createZipFile();
+        Mockito.verify(mojo).validate(mockZipFile);
+        Mockito.verify(mojo).createWarFile(mockZipFile);
+        Mockito.verify(mojo).validateWarFile(Mockito.any(WarFile.class));
+        Mockito.verify(mojo).createServerCredentials();
+        Mockito.verify(mockZipFile).close();
+        Mockito.verify(mockFile, Mockito.times(2)).exists();
+    }
+
+    /**
+     * Test validating an incorrect yaml format.
+     * @throws Exception If something is wrong
+     */
+    @Test(expected = YAMLException.class)
+    public void testInvalidYaml() throws Exception {
+        // @checkstyle StringLiteralsConcatenationCheck (6 lines)
+        final String invalid = "Some illegal Prefix\n"
+                + "Time: 2005-11-23 10:01:42 -5\n"
+                + "Admin: ed\n"
+                + "Messages:\n"
+                + "  Hello is an error information\n"
+                + "  for the configuration file\n";
+        final DeployMojo mojo = new DeployMojo();
+        mojo.validYaml(invalid);
+    }
+
+    /**
+     * Test validating an correct yaml format.
+     * @throws Exception If something is wrong
+     */
+    @Test
+    public void testValidYaml() throws Exception {
```

Comment:
@tmjee checkout this article about test method names http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names